File: Benchmarks/02-CS50x/02-Problem-sets/Pset_2.2/readability/readability.c
#include "../../cs50.h" //telling it to go up one folder and find and use the cs50.c and the cs50.h library. VERY HANDY!!!
#include <stdio.h>
#include <math.h> 
#include <ctype.h>

float calculate_L(string word);
float calculate_S(string word);



int main(void)
{  
    float index;
    int roundedup_index;

   //  take user input
    string s = get_string("Text: ");

   // plug user input inside of L and S figure out functions
   float L = calculate_L(s);
   float S = calculate_S(s);
   // get back the value of L and S from the functions

   // plug in the L and S inside of coleman liau formula
    index = 0.0588 * L - 0.296 * S - 15.8;


   // round up coleman liau value to nearest integer and call it x
    roundedup_index = lround(index);


   // output the value of x as grade x and if its 16 and above , then 16+, and if it is below 1 , then "before grade 1"

   if (roundedup_index >= 16)
   {
    printf("Grade 16+\n");
   }
   else if (roundedup_index < 1)
   {
    printf("Before grade 1\n");
   }
   else
   {
    printf("Grade: %i\n",roundedup_index);
   }
}



float calculate_L(string word)
{
    float valueL;
    int letters = 0, words = 1;

    // function
    for (int i = 0; word[i] != '\0'; i++)
    {
        if (isalpha(word[i]))
        {
            letters++;
        }
        if (word[i] == ' ')
        {
            words++;
        }
    }
    valueL = ( (float) letters / words) * 100; 
    return valueL;
}

float calculate_S(string word)
{
    float valueS;
    int sentences = 0, words = 1;

    // function
    for (int i = 0; word[i] != '\0'; i++ )
    {
        if (word[i] == ' ')
        {
            words++;
        }

        if (word[i] == '.' || word[i] == '?' || word[i] == '!')
        {
            sentences++;
        }
    }
    valueS = ( (float) sentences / words) * 100;
    return valueS;
}

File: 3_Projects_and_Labs/1_CS50x/02-Problem-sets/Pset_6.4/sentimental-readability/readability.py

# main function here
def coleman():
    user_input = input("text: ")
    L = get_L(user_input)
    S = get_S(user_input)
    
    index = round(0.0588 * L - 0.296 * S - 15.8)
    
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")
    
    
def get_L(text):
    word_list = text.split()
    words = len(word_list)
    
    letters = 0
    for char in text:
        if char.isalpha():
            letters += 1
            
    if words == 0:
        return 0
    
    return (letters/words) * 100

    
def get_S(text):
    word_list = text.split()
    words = len(word_list)
    
    sentence_counter = 0
    for char in text:
        if char in ".?!":
            sentence_counter += 1
    
    if words == 0:
        return 0
    
    return (sentence_counter/words) * 100
    
    
if __name__ == "__main__":
    coleman()